DiscordBot/Preconditions.cs

```diff
@@ -54,4 +54,30 @@ await context.Channel
             return await Task.FromResult(PreconditionResult.FromError(string.Empty));
         }
     }
+
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
+    public class RequireAutoThreadAuthorAttribute : PreconditionAttribute
+    {
+        public override async Task<PreconditionResult> CheckPermissionsAsync(ICommandContext context, CommandInfo command, IServiceProvider services)
+        {
+            var currentThread = context.Message.Channel as SocketThreadChannel;
+            if (currentThread == null) return await Task.FromResult(PreconditionResult.FromError(string.Empty));
+
+            var settings = services.GetRequiredService<Settings.Deserialized.Settings>();
+            if (!settings.AutoThreadChannels.Any(x => currentThread.ParentChannel.Id == x.Id && x.CanArchive))
+                return await Task.FromResult(PreconditionResult.FromError(string.Empty));
+
+            var user = (SocketGuildUser)context.Message.Author;
+            if (user.GuildPermissions.ManageThreads) return await Task.FromResult(PreconditionResult.FromSuccess());
+
+            var messages = await currentThread.GetPinnedMessagesAsync();
+            var firstMessage = messages.LastOrDefault();
+
+            if (firstMessage == null) return await Task.FromResult(PreconditionResult.FromError(string.Empty));
+
+            if (!firstMessage.MentionedUsers.Any(x => x.Id == context.User.Id)) return await Task.FromResult(PreconditionResult.FromError(string.Empty));
```

Comment:
If this fails, (Unlikely), it won't return a message. We may want to include something just to avoid confusing the user.

The only time this would happen is if a moderator/bot has deleted the pinned message. So it *shouldn't* but 🤷